src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_agents.clj

```diff
@@ -202,29 +201,26 @@
     (send-agent modify-instance-agent #(prime-pool! pool-context))))
 
 (schema/defn ^:always-validate
-  send-flush-and-repopulate-pool! :- jruby-schemas/JRubyPoolAgent
-  "Sends requests to the agent to flush the existing pool and create a new one."
+flush-and-repopulate-pool!
+  "Flush of the current JRuby pool. Blocks until all the instances have
+  been borrowed from the pool, and before the pool has been refilled"
```

Comment:
Ah, yeah it's weird. I'm trying to say that it will return once the pool instances have been borrowed, and that it won't wait for the instances to be destroyed and recreated.